src/core/src/core_logic/PatchInstaller.py

```diff
@@ -282,7 +333,142 @@ def install_updates(self, maintenance_window, package_manager, simulate=False):
+            package_and_dependencies = list(packages_in_batch)
+            package_and_dependency_versions = list(package_versions_in_batch)
+
+            self.include_dependencies(package_manager, packages_in_batch, all_packages, packages, package_versions, package_and_dependencies, package_and_dependency_versions)
+
+            parent_packages_installed_in_batch_count = 0
+            parent_packages_failed_in_batch_count = 0
+            number_of_dependencies_installed = 0
+            number_of_dependencies_failed = 0
+
+            code, out, exec_cmd = package_manager.install_update_and_dependencies(package_and_dependencies, package_and_dependency_versions, simulate)
+
+            for package,version in zip(package_and_dependencies, package_and_dependency_versions):
+                install_result = package_manager.get_installation_status(code, out, exec_cmd, package, version, simulate)
+
+                if install_result == Constants.FAILED:
+                    if package in packages_in_batch:
+                        # parent package
+                        self.status_handler.set_package_install_status(package_manager.get_product_name(str(package)), str(version), Constants.FAILED)
+                        self.failed_parent_package_install_count += 1
+                        patch_installation_successful = False
+                        parent_packages_failed_in_batch_count += 1
+                    else:
+                        # dependent package
+                        number_of_dependencies_failed +=1
+                elif install_result == Constants.INSTALLED:
+                    self.status_handler.set_package_install_status(package_manager.get_product_name(str(package)), str(version), Constants.INSTALLED)
+                    if package in packages_in_batch:
+                        # parent package
+                        self.successful_parent_package_install_count += 1
+                        parent_packages_installed_in_batch_count += 1
+                    else:
+                        # dependent package
+                        number_of_dependencies_installed += 1
+
+                    if package in self.last_still_needed_packages:
+                        index = self.last_still_needed_packages.index(package)
+                        self.last_still_needed_packages.pop(index)
+                        self.last_still_needed_package_versions.pop(index)
+                        installed_update_count += 1
+
+            self.attempted_parent_package_install_count += len(packages_in_batch)
+
+            # Update reboot pending status in status_handler
+            self.status_handler.set_reboot_pending(self.package_manager.is_reboot_pending())
+
+            # dependency package result management fallback (not reliable enough to be used as primary, and will be removed; remember to retain last_still_needed refresh when you do that)
```

Comment:
Where is dependency package result management implemented?

Reply:
This is being done in the method perform_status_reconciliation_conditionally.
This comment and followed by function call perform_status_reconciliation_conditionally is copied from existing code of installing packages sequentially.